src/brod.erl

```diff
@@ -248,22 +251,30 @@ get_producer(Client, Topic, Partition) ->
   brod_client:get_producer(Client, Topic, Partition).
 
 %% @equiv produce(Pid, 0, <<>>, Value)
--spec produce(pid(), binary()) ->
+-spec produce(pid(), value()) ->
                  {ok, brod_call_ref()} | {error, any()}.
 produce(Pid, Value) ->
   produce(Pid, _Key = <<>>, Value).
 
-%% @doc Produce one message. The pid should be a producer pid.
--spec produce(pid(), binary(), binary()) ->
+%% @doc Produce one message if Value is binary or iolist,
+%% or a message set if Value is a (nested) kv-list, in this case Key
+%% is discarded (only the keys in kv-list are sent to kafka).
+%% The pid should be a partition producer pid, NOT client pid.
+%% @end
+-spec produce(pid(), key(), value()) ->
         {ok, brod_call_ref()} | {error, any()}.
 produce(ProducerPid, Key, Value) ->
```

Comment:
Perhaps exclude this function from brod: api to not confuse with produce call for brod client?
If a user knows what they are doing, they can just use brod_producer directly.